### PR TITLE
changes in the demo flow to remove onbaording steps

### PR DESCRIPTION
**Pull request by bhavya2611**
This PR will remove the onboarding steps from the demo flow and make the flow same as that we had before

### CODE REVIEW

File: locales/en_US.json [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {
   "onboarding": {
+    "demo_header": "Hold on while we spin up a demo account",
```
Comment (daria-github) [inline locales/en_US.json:3]: i think the text has been changed for this — can you update to the latest? (e.g. something along the lines of "Initializing demo mode..." with no subtext

File: cypress/e2e/connect_disconnect.cy.ts [hunk 2]
```diff
@@ -129,13 +130,7 @@ describe("Disconnected Test Cases", () => {
...
   });
   it("Shows expected fields when disconnected from a wallet", () => {
-    const elements = [
-      "xmtp-logo",
-      "no-wallet-connected-header",
-      "no-wallet-connected-subheader",
-      "no-wallet-connected-cta",
-      "no-wallet-connected-subtext",
-    ];
+    const elements = ["xmtp-logo"];
```
Comment (daria-github) [inline cypress/e2e/connect_disconnect.cy.ts:133]: we can do this in a follow-on PR, but we should really add this test back outside of demo flow.

File: component-library/components/OnboardingStep/stepMapping.tsx [hunk 3]
```diff
@@ -23,6 +23,18 @@ export const stepMapping: Record<
...
 > = {
+  0: {
+    default: {
+      header: "onboarding.demo_header",
+      subheader: null,
+      cta: null,
+    },
+    loading: {
+      header: "onboarding.demo_header",
+      subheader: "onboarding.demo_subheader",
```
Comment (daria-github) [inline component-library/components/OnboardingStep/stepMapping.tsx:34]: we'll need to remove this, it displays this key in the UI while loading

File: cypress/e2e/connect_disconnect.cy.ts (hunk 2 above)
Comment (bhavya2611) [reply cypress/e2e/connect_disconnect.cy.ts:133]: We use demo flow in the tests, so for now we will have to part ways from this

Comment (daria-github) [reply cypress/e2e/connect_disconnect.cy.ts:133]: we force demo flow in the tests though, we can easily re-add this by just removing that field from local storage within the test. i don't think this test needs to be lost but doesn't need to be part of this PR

**Review by daria-github — APPROVED** [review]

**Pull request merged**